compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeSwingLayer.desktop.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:OptIn(ExperimentalComposeUiApi::class)
+
+package androidx.compose.ui.awt
+
+import androidx.compose.ui.ExperimentalComposeUiApi
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.Graphics
+import javax.accessibility.Accessible
+import javax.accessibility.AccessibleContext
+import javax.swing.SwingUtilities
+import org.jetbrains.skiko.ClipRectangle
+import org.jetbrains.skiko.ExperimentalSkikoApi
+import org.jetbrains.skiko.GraphicsApi
+import org.jetbrains.skiko.SkiaLayerAnalytics
+import org.jetbrains.skiko.swing.SkiaSwingLayer
+import org.jetbrains.skiko.swing.SkiaSwingLayerComponent
+
+/**
+ * Provides [component] that can be used as a Swing component.
+ * Content set in [setContent] will be drawn on this [component].
+ *
+ * [ComposeSwingLayer] provides smooth integration with Swing, so z-ordering, double-buffering etc. from Swing will be taken into account.
+ *
+ * However, if smooth interop with Swing is not needed, consider using [androidx.compose.ui.awt.ComposeWindowLayer]
+ */
+internal class ComposeSwingLayer(
```

Comment:
@m-sasha I agree about names, I don't like it too.
WDYT about:
ComposeLayer -- keep as historical one. Or change to ComposeSkiaBridge?
ComposeSwingLayer -- keep ComposeSwingLayer
ComposeSwingSkiaLayer -- change to ComposeSwingLayerComponent?
ComposeWindowLayer -- keep ComposeWindowLayer
ComposeWindowSkiaLayer -- change to ComposeWindowSkiaComponent?
SkiaLayer -- keep as historical one
SkiaSwingLayer -- keep
SkiaSwingLayerComponent -- remove this

Reply:
I vote for this:
```
ComposeLayer -> ComposeBridge
ComposeSwingLayer -> SwingComposeBridge
ComposeSwingSkiaLayer -> SwingComposeBridge.<anonymous field object>
ComposeWindowLayer -> WindowComposeBridge
ComposeWindowSkiaLayer -> WindowComposeBridge.<anonymous field object>
```

- subclasses better name in a format `<prefix> + <base class>`
- `ComposeSkiaBridge` is better to understand, but `WindowComposeSkiaBridge` is too long

anonymous field object:
<img width="540" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/fa02c44e-a52a-4028-988d-ab2dd75242eb">

Reply:
That sounds better.

Reply:
Agree, I also thought about such namings with `Bridge` suffix